docs/getting-started/server/guide.md

```diff
@@ -294,6 +294,29 @@ attempting to "Build and Run the Server".
 
 </Bitwarden>
 
+## Configure NuGet with GitHub Packages
+
+Server-side projects and solutions may use
+[Bitwarden-shared .NET extension libraries hosted on GitHub Packages](https://github.com/orgs/bitwarden/packages?repo_name=dotnet-extensions)
+and these packages require authentication for access.
+
+First, [generate](https://github.com/settings/tokens/new) a GitHub personal access token (classic)
+with the `packages:read` scope only. You can set an expiration date but it may be easier to leave it
+without one considering the scope. Copy the token value and run:
+
+```bash
+IFS= read -rs GITHUB_PAT < /dev/tty
+```
+
+along with pasting the value and pressing Enter. Next, run:
+
+```bash
+dotnet nuget add source --username bitwarden --password $GITHUB_PAT --store-password-in-clear-text --name github --configfile ~/.nuget/NuGet/NuGet.Config "https://nuget.pkg.github.com/bitwarden/index.json"
```

Comment:
I think we should add a `nuget.config` file with `<add key="github" value="https://nuget.pkg.github.com/bitwarden/index.json" />` in it so that the url is in the repo and doesn't have to be added here. 

Instead the command they run will be: 
```
dotnet nuget update source github --username bitwarden --password $GITHUB_PAT --store-password-in-clear-text --configfile ~/.nuget/NuGet/NuGet.Config
```

This way there will be a more informative error when `dotnet restore` ultimately fails:
```
/usr/share/dotnet/sdk/8.0.101/NuGet.targets(156,5): warning : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured. [/workspace/src/Api/Api.csproj]
/workspace/src/Api/Api.csproj : error NU1301: Failed to retrieve information about 'Bitwarden.Extensions.Hosting' from remote source 'https://nuget.pkg.github.com/bitwarden/download/bitwarden.extensions.hosting/index.json'.
```

Also we will have to add a command that works on windows and we should probably link [here](https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#config-file-locations-and-uses) somewhere in the docs. As the location can differ based on installation method.